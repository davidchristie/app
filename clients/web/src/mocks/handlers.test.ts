interface Handler {
  method?: "DELETE" | "GET" | "POST";
  url: string;
}

const handlers: Handler[] = [
  {
    url: "/api/v1/auth/session",
  },
];

handlers.forEach(({ method = "GET", url }) => {
  describe(`${method} ${url}`, () => {
    it("responds correctly", async () => {
      const response = await fetch("/api/v1/auth/session");
      const body = await response.json();
      expect(response.ok).toBe(true);
      expect(response.status).toBe(200);
      expect(body).toMatchSnapshot();
    });
  });
});
