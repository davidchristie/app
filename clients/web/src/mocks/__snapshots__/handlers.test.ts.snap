// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/v1/auth/session responds correctly 1`] = `
Object {
  "user": Object {
    "email": "test_user@email.com",
    "id": "97406d59-7a49-4f1e-bb79-aba34cfcb405",
    "name": "Test User",
  },
}
`;
